libarchive/archive_read_disk_posix.c

```diff
@@ -666,10 +666,8 @@ setup_suitable_read_buffer(struct archive_read_disk *a)
 		/*
 		 * Calculate proper address for the filesystem.
 		 */
-		s = (uintptr_t)cf->allocation_ptr;
+		s = (uintptr_t)cf->allocation_ptr + xfer_align;
```

Comment:
Note:  Adding `xfer_align` makes no difference to the result in the next line.   Generally:   `((A + B) % B) == (A % B)`

Reply:
You could avoid the branch here by changing the update below to
```
cf->buff = cf->allocation_ptr + xfer_align - s;
```
In the case where `cf->allocation_ptr` is already aligned, that would waste the first block in the allocated buffer.  The previous logic did not do this. 

I'd be in favor of this if it made the code easier to understand, but avoiding one branch here is not really that important.